### PR TITLE
Add/jitm plans page nofification

### PR DESCRIPTION
**Pull request by ayebare**
Related to #
pe0hvg-4y-p2#comment-246

## Proposed Changes
Add a Limited time offer JITM nudge to the plans pate.
**Notification text:**
48-hour flash sale—save 30% off all annual plans! Use coupon code {couponCode} at checkout.

**Button text**
[Save now]

**Clicked state**
Discount coupon applied! Select your plan below and check your final discounted price at checkout.

https://github.com/Automattic/wp-calypso/assets/11347472/7b114e3f-3a87-4ab9-ab5c-d414bbc28f5a

## Testing Instructions
Find testing instructions here D121094-code

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by mpkelly — COMMENTED** [review]
Code looks fine. I wasn't able to complete the test plan and have commented on the diff.

File: client/state/data-layer/wpcom/sites/jitm/index.js [hunk 1]
```diff
@@ -36,6 +36,9 @@ const transformApiRequest = ( { data: jitms } ) =>
 		CTA: {
 			message: unescapeDecimalEntities( jitm.CTA.message ),
 			link: unescapeDecimalEntities( jitm.CTA.link || '' ),
+			target: unescapeDecimalEntities(
+				jitm.CTA.target || '' === jitm.CTA.target ? jitm.CTA.target : '_blank'
```
Comment (gcsecsey) [inline client/state/data-layer/wpcom/sites/jitm/index.js:40]: `''` is falsy, so the ternary isn't necessary in this case:
```suggestion
				jitm.CTA.target || '_blank'
```

**Review by gcsecsey — APPROVED** [review]
Changes LGTM, just added a minor comment. With the latest test steps on D121094-code, I could see the newly added messages:

CleanShot 2023-10-16 at 11 39 55@2x

**Review by gcsecsey — APPROVED** [review]
Changes LGTM, just added a minor comment. With the latest test steps on D121094-code, I could see the newly added messages:

CleanShot 2023-10-16 at 11 39 55@2x

File: client/state/data-layer/wpcom/sites/jitm/index.js (hunk 1 above)
Comment (ayebare) [reply client/state/data-layer/wpcom/sites/jitm/index.js:40]: Thanks @gcsecsey, we don't want to return '_blank' when "target" is an empty string. In some situations especially when we want to load content without a full page reload, having a target forces a refresh.

Comment (gcsecsey) [reply client/state/data-layer/wpcom/sites/jitm/index.js:40]: Oh I see, thanks for clarifying!

**Pull request merged**